sql/analyzer/rules.go

```diff
@@ -27,6 +27,7 @@ var DefaultRules = []Rule{
 var OnceBeforeDefault = []Rule{
 	{"resolve_subqueries", resolveSubqueries},
 	{"resolve_tables", resolveTables},
+	{"check_aliases", checkAliases},
```

Comment:
this should be a validation rule

Reply:
validation rules are executed after regular rules, this must executed with the parsed plan before transformations

Reply:
But it's a validation rule. Why can't it be run in validation phase?

Reply:
Since validation phase runs after the default rules, the plan to be validated could be modified enough to not keep this alias case.

An alias defined and used in the same projection in the original query, it couldn't be there in the plan after the transformations (reorder/remove/pushdown/joins...) (that's the case of the issue).

To be sure that a query which is making this is not allowed, the check must be done on the parsed plan or on a plan near to it.